prody/proteins/interactions.py

```diff
@@ -3010,6 +3239,484 @@ def showSminaTermValues(data):
+    
+    Before using the function, install Foldseek:
+    >>> conda install bioconda::foldseek
+    More information can be found:
+    https://github.com/steineggerlab/foldseek?tab=readme-ov-file#databasesand
+
+    This function will not work under Windows.
+    
+    :arg pdb_file: A PDB file path
+    :type pdb_file: str
+
+    :arg chain: Chain identifier
+    :type chain: str
+
+    :arg coverage_threshold: Coverage threshold 
+            by default 0.3
+    :type coverage_threshold: float
+
+    :arg tm_threshold: TM-score threshold
+            by default 0.5
+    :type tm_threshold: float
+    
+    :arg database_folder: Folder with the database
+            by default '~/Donwload/foldseek/pdb'
+            To download the database use: 'foldseek databases PDB pdb tmp' in the console 
+    :type database_folder: str
+    """
+    
+    import os
+    import subprocess
+    import re
+    import sys
+
+    database_folder = kwargs.pop('database_folder', '~/Donwload/foldseek/pdb')
+    coverage_threshold = kwargs.pop('coverage_threshold', 0.3)
+    tm_threshold = kwargs.pop('tm_threshold', 0.5)    
+    
+    # Define the amino acid conversion function
+    def aa_onelet(three_letter_code):
+        codes = {
+            'ALA': 'A', 'ARG': 'R', 'ASN': 'N', 'ASP': 'D', 'CYS': 'C',
+            'GLN': 'Q', 'GLU': 'E', 'GLY': 'G', 'HIS': 'H', 'ILE': 'I',
+            'LEU': 'L', 'LYS': 'K', 'MET': 'M', 'PHE': 'F', 'PRO': 'P',
+            'SER': 'S', 'THR': 'T', 'TRP': 'W', 'TYR': 'Y', 'VAL': 'V',
+            'MSE': 'M'
+        }
+        return codes.get(three_letter_code)
+
+    # Function to extract the sequence from the PDB file
+    def extract_sequence_from_pdb(pdb_file, chain, output_file):
+        sequence = []
+        prev_resid = -9999
+
+        with open(pdb_file, 'r') as pdb:
+            for line in pdb:
+                if line.startswith("ATOM"):
+                    ch = line[21]
+                    resid = int(line[22:26].strip())
+                    aa = line[17:20].strip()
+
+                    if ch == chain and resid != prev_resid:
+                        one_aa = aa_onelet(aa)
+                        if one_aa:
+                            sequence.append(one_aa)
+                            prev_resid = resid
+
+        with open(output_file, 'w') as seq_file:
+            seq_file.write(''.join(sequence))
+
+    # Inputs
+    fpath = pdb_file.strip()
+    chain = chain.strip()
+    cov_threshold = float(coverage_threshold)
+    tm_threshold = float(tm_threshold)
+    
+    # Filter PDB file
+    awk_command = "awk '{{if (substr($0, 22, 1) == \"{0}\") print}}'".format(chain)
+    subprocess.run("cat {0} | grep '^ATOM' | {1} > inp.pdb".format(fpath, awk_command), shell=True)
+
+    # Run foldseek and other commands
+    subprocess.run([
+        'foldseek', 'easy-search', 'inp.pdb', database_folder, 'prot.foldseek',
```

Comment:
we should probably check that that the database_folder exists before running foldseek and having it fail if it doesn't

Also, how are we handling errors from all these subprocess commands?

Reply:
I added that, but I don't know which problems we might have with the subprocess. Maybe Anupam can say something more about it?

Reply:
@jamesmkrieger and @karolamik13 I just discussed with Anupam about this PR. I will test and make sure that a few checks are in place for these subprocesses.

Reply:
Thanks Anthony!

Reply:
Ok. Great. Thank you!

Reply:
@jamesmkrieger and @karolamik13 I have tested this with ADK and everything is working as expected (except one small thing with the plotting, which I am debugging now). I added in some checks to make sure the subprocesses are covered. Once I am done, I will open a PR for your branch, Karolina and you can have a look at my changes. There was a bug I found and fixes, so that change is necessary to merge in, the other checks are helpful but not essential.

Reply:
Thank you!